frontend/src/app/guards/admin.guard.ts

```diff
@@ -0,0 +1,22 @@
+import { CanActivate, ActivatedRouteSnapshot, RouterStateSnapshot, Router, UrlTree } from '@angular/router';
+import { Injectable } from '@angular/core';
+import { SessionService } from '../services/session.service';
+import { Observable } from 'rxjs';
+
+@Injectable()
+export class AdminGuard implements CanActivate {
+
+    constructor(private session: SessionService, private router: Router) {}
+
+    canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): boolean | UrlTree | Observable<boolean | UrlTree> | Promise<boolean | UrlTree> {
+        const isAdmin = this.session.getUserData().role;
```

Comment:
vagy már ott vizsgált hogy admin -e 
```suggestion
        const isAdmin = this.session.getUserData().role === 'ADMIN';
```
vagy ne ez legyen a változó neve
```suggestion
        const userRole = this.session.getUserData().role;
```